### ISSUE TITLE
Feature: Capture video from an indoor station

### ISSUE THREAD

**Issue by pergolafabio** [pergolafabio/Hikvision-Addons#162]
### Is there an existing feature request for this?

- [X] I have searched the existing issues

### Add-on

Doorbell

### Your feature request

Would be nice if we could capture the video from the indoor station, useful for people with no access to outdoor stations (apperments), but they do have access to indoor, since those devices do have LAN+Wifi...

**Comment by PKort** [comment]
You can capture the video from the station in parallel to this AddOn. I used go2RTC for this. I can also get 2-way audio from the station in HA.
For details see here: https://github.com/PKort/HA_HikVision

**Comment by pergolafabio** [comment]
same as : 

https://github.com/pergolafabio/Hikvision-Addons/blob/main/advanced_card/twowayaudio.with.advanced.camera.card.md


 :-)

**Comment by tach2004** [comment]
Hello. First of all, thank you very much for this great add-on. It helped me a lot to integrate my Metzler VDM10 2.0 (which is the same as the Hikvision DS-KD8003) into Home Assistant 🙂
I also successfully integrated the motion sensor via an MQTT binary sensor, and it works perfectly.

What is still missing, of course, would be an image or, even better, a live camera stream. That’s why I wanted to ask in this post whether something like this already exists or is planned for the future. That would be absolutely awesome, especially without having to use Hikvision’s cloud integration, which I definitely want to avoid.

I also have two additional indoor stations, in case that is helpful 🙂

Thanks again, and I’m really looking forward to your reply!

**Comment by PKort** [comment]
> Hello. First of all, thank you very much for this great add-on. It helped me a lot to integrate my Metzler VDM10 2.0 (which is the same as the Hikvision DS-KD8003) into Home Assistant 🙂 I also successfully integrated the motion sensor via an MQTT binary sensor, and it works perfectly.
> 
> What is still missing, of course, would be an image or, even better, a live camera stream. That’s why I wanted to ask in this post whether something like this already exists or is planned for the future. That would be absolutely awesome, especially without having to use Hikvision’s cloud integration, which I definitely want to avoid.
> 
> I also have two additional indoor stations, in case that is helpful 🙂
> 
> Thanks again, and I’m really looking forward to your reply!

You can use my development as starting point - you can find instructions here:
https://github.com/PKort/HA_HikVision
Also there was a page with similar development that @pergolafabio shared some time ago, but I just can't find it anymore. Maybe he will share it as well.

**Comment by pergolafabio** [comment]
> Hello. First of all, thank you very much for this great add-on. It helped me a lot to integrate my Metzler VDM10 2.0 (which is the same as the Hikvision DS-KD8003) into Home Assistant 🙂 I also successfully integrated the motion sensor via an MQTT binary sensor, and it works perfectly.
> 
> What is still missing, of course, would be an image or, even better, a live camera stream. That’s why I wanted to ask in this post whether something like this already exists or is planned for the future. That would be absolutely awesome, especially without having to use Hikvision’s cloud integration, which I definitely want to avoid.
> 
> I also have two additional indoor stations, in case that is helpful 🙂
> 
> Thanks again, and I’m really looking forward to your reply!

If you have access to your outdoor station, just use the RTSP link, no need to capture it from the indoor...

**Comment by pergolafabio** [comment]
Is there someone who doesn't have connection to the outdoor station by it's IP address (appartement complex) , I maybe have an option to capture the video from the indoor station, so it internally connects to the outdoor station

**Comment by kzaoaai** [comment]
I would be interested in trying that. I do have wifi access to the outdoor station, but it's not in optimal wifi range.

**Comment by pergolafabio** [comment]
> I would be interested in trying that. I do have wifi access to the outdoor station, but it's not in optimal wifi range.

Do you have the admin access to indoor panel?  Does my addon work for you if you add the indoor panel to the addon config?

**Comment by kzaoaai** [comment]
Yes and yes, my DS-KH6350 is added into HA.

**Comment by pergolafabio** [comment]
what is your OUTDOOR IP, i have it for now hardcoded in the config, so the indoor station connects to the outdoor station internally
but im already testing woith another user right now

**Comment by kzaoaai** [comment]
I’m guessing you mean the LAN ip. Door station IP is 192.0.0.65

**Comment by pergolafabio** [comment]
yes, correct

**Comment by pergolafabio** [comment]
im uploading beta 95 , there is an optional config now for the indoor station to point to outdoor IP, like below...

```
doorbells:
  - name: DS-KH9310
    ip: 192.168.0.72
    username: admin
    password: xxxx
    outdoor_ip: 192.168.0.70
system:
  log_level: INFO
  sdk_log_level: NONE
mqtt: {}
```

then there will be a new button 'take sbapshot'

<img width="389" height="109" alt="Image" src="https://github.com/user-attachments/assets/080237c6-0044-49d5-9a18-23646c706eaa" />

with some hope, it should display:

```
2026-01-14 09:59:52.424 | INFO     | mqtt_input:_take_snapshot_callback:448 - Snapshot triggered for DS-KH9310. Connecting to Outdoor Station on IP: 192.168.0.70
2026-01-14 09:59:52.603 | INFO     | mqtt_input:_take_snapshot_callback:504 - SUCCESS: Snapshot saved to /media/ds-kh9310/snapshot_20260114_095952.jpg
```

**Comment by harisma-git** [comment]
It works!

**Comment by harisma-git** [comment]
Is it possible to have the video stream as well?

**Comment by pergolafabio** [comment]
thats good news! so that also means your outdoor station has the same admin password as your indoor station?
Do you have direct connection to your outdoor as well ? or is this on a private network?

**Comment by pergolafabio** [comment]
Also what indoor device do you have?
Can you test on your indoor device config, for the "outdoor_ip" enter there not the outdoor. But actually the same indoor IP... I want to know if you are able to grab a picture from the indoor panel too, since it's a linked device anyway... My indoor panel doesnt work with that, maybe yours will work

**Comment by kzaoaai** [comment]
how did you install the beta?

<img width="586" height="225" alt="Image" src="https://github.com/user-attachments/assets/ef3c4069-48b5-4811-8613-d38db8bf53b1" />

```2026-01-14 15:53:37.189 INFO (MainThread) [supervisor.docker.interface] Downloading docker image ghcr.io/pergolafabio/hikvision-doorbell with tag 3.1.
2026-01-14 15:53:38.390 ERROR (MainThread) [supervisor.docker.interface] Can't install ghcr.io/pergolafabio/hikvision-doorbell:3.1: DockerError(404, 'manifest unknown')
2026-01-14 15:53:38.391 ERROR (MainThread) [supervisor.addons.addon] Could not pull image to update addon aff2db71_hikvision_doorbell_beta: Can't install ghcr.io/pergolafabio/hikvision-doorbell:3.1: DockerError(404, 'manifest unknown')```

**Comment by pergolafabio** [comment]
Try to uninstall all and start clean

**Comment by kzaoaai** [comment]
Tried that. Even removed the addon, cleared data and rebooted the HA machine. I also removed the repo and re-added. Still same error.

**Comment by pergolafabio** [comment]
Strange. Do you actually see beta 95 ? I don't see that in your log/screenshot

**Comment by kzaoaai** [comment]
The beta addon says 3.1

<img width="691" height="140" alt="Image" src="https://github.com/user-attachments/assets/25453a87-83bb-44b6-b1d7-57ea8347154d" />

<img width="1008" height="295" alt="Image" src="https://github.com/user-attachments/assets/b2b36145-5bc2-45b8-8688-cc00e8b3d94e" />

**Comment by pergolafabio** [comment]
thats old, maybe some cache, that doesnt exist anymore, should look like below:

<img width="872" height="159" alt="Image" src="https://github.com/user-attachments/assets/19223bef-79f1-4404-ba20-1f3b973496ee" />

<img width="1073" height="207" alt="Image" src="https://github.com/user-attachments/assets/5a6db523-4ac1-4c04-b1ba-6c131bc85eed" />

<img width="1045" height="233" alt="Image" src="https://github.com/user-attachments/assets/628507fe-d608-4650-951c-c0cd034cc1bf" />

**Comment by kzaoaai** [comment]
it's weird tried everything and it still shows 3.1. I even forked the repo and added my own, same result. I will try the docker container method later.

**Comment by pergolafabio** [comment]
> It works!

Hey, the code inititates a login to the outdoor_ip with the same admin credentials thay you used for the indoor panel
am i correct saying that the outdoor panel has the same admin password? can you also login in to the outdoor panel directly? or is it on another network?

if you live in apperment with no network access to outdoor station, its strange that they setup the same admin passsword? 


not sure if this feature is actually usefull.. i need to find a way to grab picture from indoor panel directly, without outdoor_ip

**Comment by harisma-git** [comment]
> > It works!
> 
> Hey, the code inititates a login to the outdoor_ip with the same admin credentials thay you used for the indoor panel am i correct saying that the outdoor panel has the same admin password? can you also login in to the outdoor panel directly? or is it on another network?
> 
> if you live in apperment with no network access to outdoor station, its strange that they setup the same admin passsword?
> 
> not sure if this feature is actually usefull.. i need to find a way to grab picture from indoor panel directly, without outdoor_ip

Yes, the outdoor panel uses the same admin password.

In my first apartment, I also have direct network access to the outdoor panel and everything works as expected (I can take a snapshot from the indoor panel).

In my second apartment (without direct connection to the outdoor panel), however, the add-on logs show:
"Outdoor IP not configured for doorbell: Building ..."

This is really strange, because the configuration is the same and the outdoor_ip is definitely configured.

Do you have any idea what could be causing this?

I believe that in 99% of installations the indoor and outdoor panels use the same admin password, since they are usually configured by the same installer.

When it works correctly, this feature is actually very useful.

Of course, being able to retrieve the snapshot directly from the indoor panel only would be even better.

**Comment by pergolafabio** [comment]
Yeah, I was looking for a way to take it from indoor, but I can't get it to work :-(

Not sure about your second appartement, but I'm cleaning up code and will release a new beta next week, so you can test again .. but I will leave then the outdoor_ip in the config...

**Comment by harisma-git** [comment]
My indoor panel is a DS-KH6350, and I can’t test using the same IP because I get the error I mentioned earlier.

**Comment by pergolafabio** [comment]
Yeah, using same IP for outdoor doesn't work indeed...
Anyway, I'm trying to find a way to lookup the IP of the outdoor linked to indoor,  so there is no need to add it to the config...

**Comment by harisma-git** [comment]
Really really strange!

<img width="995" height="48" alt="Image" src="https://github.com/user-attachments/assets/c7f8fe89-015f-4a9d-9999-6f6bd0708ef5" />

**Comment by pergolafabio** [comment]
And can you show me the addon config in yaml?

**Comment by harisma-git** [comment]
<img width="276" height="248" alt="Image" src="https://github.com/user-attachments/assets/c1ebbfa5-3e8c-4554-83b0-e7a8bd66c9a6" />

**Comment by pergolafabio** [comment]
Strange indeed, but I try to figure out a way to retrieve the linked outdoor IP automatically, so there is no need for it to be in the config...

Best approach is offcourse to snap it directly from indoor panel...

**Comment by harisma-git** [comment]
Do you think it will be possible to capture real-time video in the future?

**Comment by pergolafabio** [comment]
From indoor?? I can't even grab a picture ;-)

If that would work. Next step is indeed video

**Comment by kzaoaai** [comment]
> <img alt="Image" width="276" height="248" src="https://private-user-images.githubusercontent.com/51525877/536419919-c1ebbfa5-3e8c-4554-83b0-e7a8bd66c9a6.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njg1MTc5MzAsIm5iZiI6MTc2ODUxNzYzMCwicGF0aCI6Ii81MTUyNTg3Ny81MzY0MTk5MTktYzFlYmJmYTUtM2U4Yy00NTU0LTgzYjAtZTdhOGJkNjZjOWE2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTE1VDIyNTM1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUwYzlhYjBhMjk3MWVkZjlmNGEyZGU0NDdlNmUyMjY4OTJiMWMxMTc5NWFjOTQ5NGJjNDRlNjM2NGIyZTU0MzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.gf1sV-KifqbPH3eoLGXnytQZGdUIJCLUX4FSDnxXLqY">

I thought outdoor_ip should go in the indoor panel section. This is what the error is complaining about.

**Comment by kzaoaai** [comment]
I spun up a beta docker container to test. I tried 4 different settings for "outdoor_ip":
1- External doorbell's LAN IP -> fails with this error "Login Failed for 192.0.0.65: Error 10"
2- External doorbell's WIFI IP -> success (but pointless for those with WIFI access to the external doorbell)
3- Internal panel's LAN IP -> fails with same error as # 1
4- Internal panel's WIFI IP -> fails with error "SDK FAILURE: Error 29 during capture"

**Comment by pergolafabio** [comment]
Yeah, your admin password on outdoor is different , that's why it fails... I need to find a way to grab it from indoor directly

**Comment by pergolafabio** [comment]
i added beta 96 , no need to configure outdoor_ip anymore in the config ... im trying to test it from indoor directly ... can you guys test ? if you have access to outdoor station, test it too please, but outdoor should work
im interested if it works from indoor panel

I have an ds-kh95xx device, for me it doesnt work, maybe for yours it will , since kh63 is totally different...

**Comment by pergolafabio** [comment]
can you guys also download the SDK from here:
https://www.hikvision.com/en/support/tools/hitools/?type=IP

in the lib folder there is a client tool ... i also receive error 23 there when i try to capture from indoor, what about you?

<img width="1004" height="766" alt="Image" src="https://github.com/user-attachments/assets/00ae8ffd-3f07-49e4-8b50-a00e800d87b0" />

**Comment by harisma-git** [comment]
<img width="1005" height="511" alt="Image" src="https://github.com/user-attachments/assets/5fdcdcaa-4eb9-4d2f-a77b-996812f777e8" />

**Comment by pergolafabio** [comment]
Not good :-(

**Comment by harisma-git** [comment]
Is there any way to bypass the indoor capture and use the outdoor_ip ?

**Comment by pepsonEL** [comment]
> i added beta 96 , no need to configure outdoor_ip anymore in the config ... im trying to test it from indoor directly ... can you guys test ? if you have access to outdoor station, test it too please, but outdoor should work im interested if it works from indoor panel
> 
> I have an ds-kh95xx device, for me it doesnt work, maybe for yours it will , since kh63 is totally different...

How install beta version ? I see only official.

**Comment by pergolafabio** [comment]
> Is there any way to bypass the indoor capture and use the outdoor_ip ?

My next approach is look for an internal connection with sdk, I want to avoid the login to outdoor_ip , that will not work for everyone when admin pass is different

**Comment by pergolafabio** [comment]
> > i added beta 96 , no need to configure outdoor_ip anymore in the config ... im trying to test it from indoor directly ... can you guys test ? if you have access to outdoor station, test it too please, but outdoor should work im interested if it works from indoor panel
> > I have an ds-kh95xx device, for me it doesnt work, maybe for yours it will , since kh63 is totally different...
> 
> How install beta version ? I see only official.

It's another addon. On same repository, you can't update stable version to beta...
Make sure also to stop the stable addon, when you run the beta

**Comment by pepsonEL** [comment]
<img width="1037" height="394" alt="Image" src="https://github.com/user-attachments/assets/7879314a-1698-4a01-a3d6-27f0d1aeb020" />

But i have only version like 3.1

**Comment by pergolafabio** [comment]
You probably still see old repo. Remove that , refresh, add again

**Comment by pepsonEL** [comment]
I add again but still i see the same., But Official stable i see correct 3.0.26

**Comment by pergolafabio** [comment]
should look like:

stable indeed : 3.0.26
beta => Current version: 3.0.0-beta.96

do also a "check for addons" manually, wait a min, then refresh 

<img width="919" height="164" alt="Image" src="https://github.com/user-attachments/assets/47b8dd5b-0410-4285-bd38-5253333912df" />

**Comment by harisma-git** [comment]
> > Is there any way to bypass the indoor capture and use the outdoor_ip ?
> 
> My next approach is look for an internal connection with sdk, I want to avoid the login to outdoor_ip , that will not work for everyone when admin pass is different

I understand it, but give the option for those of us for whom the code is the same and it works :)

**Comment by pepsonEL** [comment]
Still show the same

<img width="1166" height="149" alt="Image" src="https://github.com/user-attachments/assets/a31bedfe-f790-4a52-ba15-4e0bf07ba641" />

To repository link it is this ? https://github.com/pergolafabio/Hikvision-Addons

**Comment by pergolafabio** [comment]
yes

remove it, do a check for addons updates, mabye restart HA
then add repo again, it must be there , see link:
https://github.com/pergolafabio/Hikvision-Addons/blob/main/doorbell_beta/config.yaml

**Comment by pepsonEL** [comment]
Your link repo is wrong:

https://github.com/pergolafabio/Hikvision-Addons/blob/main/doorbell_beta/config.yaml

**Comment by pergolafabio** [comment]
No , it's not

**Comment by pergolafabio** [comment]
ahh, i see it now ...

**Comment by pergolafabio** [comment]
just add it as custom, dont click on link, try it, it will work

**Comment by pepsonEL** [comment]
OK i know but when try add custom and put your link it is wrong
Only can add this link: https://github.com/pergolafabio/Hikvision-Addons

But still no show beta version as you show.

**Comment by pergolafabio** [comment]
No id then, something wrong on your side

**Comment by kzaoaai** [comment]
I had the same issue with installing the addon. I just tried again by removing the repo, and I get this error when re-adding it
`Cmd('git') failed due to: exit code(128) cmdline: git clone -v --recursive --depth=1 --shallow-submodules -- https://github.com/pergolafabio/Hikvision-Addons/blob/main/doorbell_beta/config.yaml /data/addons/git/f117d17b stderr: 'Cloning into '/data/addons/git/f117d17b'... fatal: repository 'https://github.com/pergolafabio/Hikvision-Addons/blob/main/doorbell_beta/config.yaml/' not found '
`

Docker install works. Just tested version 96. I am getting the same errors as the other tester. I will download the SDK and see what happens.

**Comment by pergolafabio** [comment]
strange , this file exists ...
https://github.com/pergolafabio/Hikvision-Addons/blob/main/doorbell_beta/config.yaml

**Comment by pergolafabio** [comment]
as a test, uninstalled my addons, removed repo, check update
added repo again:  https://github.com/pergolafabio/Hikvision-Addons 

no issue  :)

**Comment by pergolafabio** [comment]
i may have find a way, need to test next week, i wiresharked an live view press on the indoor panel, its doing an rtsp stream to the outdoor station  , but using 10010110001 as username, thats actually the SIP number (private hik)
Seems the password is the same ... trying with VLC doesnt work, maybe the outdoor station only accepts it from the linked devices only

need to implement this in the addon ... if it works, then live view should be possible too ...

```
RTSP/1.0 401 Unauthorized
CSeq: 2
WWW-Authenticate: Digest realm="Hikvision", nonce="xxxxx", random="+462w9mVKGL+MpScHTtAHnUz0v2OqIBWK69eQZlKqyX7ypxIT3K5fZHItr4Y22mW", stale="FALSE"
WWW-Authenticate: Basic realm="Hikvision"
Date:  Tue, Dec 28 2021 12:29:27 GMT


DESCRIBE rtsp://192.168.0.70/streaming/channels/101 RTSP/1.0
CSeq: 3
Accept: application/sdp
Authorization: Digest username="10010110001", realm="Hikvision", nonce="xxxx", uri="/streaming/channels/101", response="xxxxxx"
User-Agent: HIKVISION player NVR
```

**Comment by pepsonEL** [comment]
> as a test, uninstalled my addons, removed repo, check update added repo again: https://github.com/pergolafabio/Hikvision-Addons
> 
> no issue :)

I also done this but no show beta 96

**Comment by kzaoaai** [comment]
> i may have find a way, need to test next week, i wiresharked an live view press on the indoor panel, its doing an rtsp stream to the outdoor station , but using 10010110001 as username, thats actually the SIP number (private hik) Seems the password is the same ... trying with VLC doesnt work, maybe the outdoor station only accepts it from the linked devices only
> 
> need to implement this in the addon ... if it works, then live view should be possible too ...
> 
> ```
> RTSP/1.0 401 Unauthorized
> CSeq: 2
> WWW-Authenticate: Digest realm="Hikvision", nonce="xxxxx", random="+462w9mVKGL+MpScHTtAHnUz0v2OqIBWK69eQZlKqyX7ypxIT3K5fZHItr4Y22mW", stale="FALSE"
> WWW-Authenticate: Basic realm="Hikvision"
> Date:  Tue, Dec 28 2021 12:29:27 GMT
> 
> 
> DESCRIBE rtsp://192.168.0.70/streaming/channels/101 RTSP/1.0
> CSeq: 3
> Accept: application/sdp
> Authorization: Digest username="10010110001", realm="Hikvision", nonce="xxxx", uri="/streaming/channels/101", response="xxxxxx"
> User-Agent: HIKVISION player NVR
> ```

Hopefully this works. I tried with the SDK and got the same error23 trying to stream from indoor panel.

**Comment by pergolafabio** [comment]
Tried some first tests already. But always fails with 401 :-(

**Comment by pepsonEL** [comment]
Look when try install beta addon which me show version 3.1 in supervisor Home assistant show me error in logs:
2026-01-18 19:26:19.537 ERROR (MainThread) [supervisor.docker.interface] Can't install ghcr.io/pergolafabio/hikvision-doorbell:3.1: DockerError(404, 'manifest unknown')
2026-01-18 19:26:19.538 ERROR (MainThread) [supervisor.addons.addon] Could not pull image to update addon aff2db71_hikvision_doorbell_beta: Can't install ghcr.io/pergolafabio/hikvision-doorbell:3.1: DockerError(404, 'manifest unknown')

**Comment by pergolafabio** [comment]
3.1 isn't on my repository, no idea why your HA wants to download it, look at the root of my github

**Comment by pergolafabio** [comment]
hey @harisma-git , can you try beta 98
the outdoor_ip is not needed anymore in the addon config, i added a function , if you press the "take snapshot" button from an indoor, it will first automaticly find the linked outdoor IP, then use that IP to login

can you try?

**Comment by pergolafabio** [comment]
for me:

```
2026-01-22 21:55:07.144 | INFO     | mqtt_input:_take_snapshot_callback:436 - Received take snapshot command, doorbell: DS-KH9310
2026-01-22 21:55:07.145 | INFO     | doorbell:take_snapshot:235 - Indoor station detected, fetching linked Outdoor IP for snapshot...
2026-01-22 21:55:07.186 | INFO     | doorbell:get_outdoor_ip:219 - Found linked Door Station IP: 192.168.0.70
2026-01-22 21:55:07.551 | INFO     | doorbell:_save_snapshot_result:316 - Snapshot saved: /media/ds-kh9310/snapshot_20260122_215507.jpg
```

**Comment by harisma-git** [comment]
Hi @pergolafabio . The username/password and the ip are 100% correct.

<img width="1004" height="211" alt="Image" src="https://github.com/user-attachments/assets/e7289db0-311d-4708-bb9a-0fdfeaf1cc0e" />

**Comment by pergolafabio** [comment]
Hmm. Error 7 is a network failure, is the IP correct that was dynamicly found now? This error is not related to user/pass

**Comment by pergolafabio** [comment]
Hmm, IP is correct you said... Not sure why it doesn't work, my guess it's more a local issue?

**Comment by harisma-git** [comment]
Yeah it is correct. The strange thing is that the intercom works fine on its own, but I found it offline in the Hik-Connect app....By the way, it’s configured to log in using the same credentials. Why not use separate ones? They can still be different and someone can still know them, or ask the installer.

**Comment by pergolafabio** [comment]
Yeah I know, step by step :-)
We will see if it's needed or not. Don't think most people have the admin anyway, ik see if I get requests for this... I don't want to make the config to complicated... If the admin of the outdoor was different because the installed configured it like this. Then it's better to have the same admin password on indoor too

**Comment by pergolafabio** [comment]
> > as a test, uninstalled my addons, removed repo, check update added repo again: https://github.com/pergolafabio/Hikvision-Addons
> > no issue :)
> 
> I also done this but no show beta 96

hey,i think i know why you were still seeing 3.1 , was indeed an folder still present

can you uninstall addons, remove repository from the addons, refresh, and repository again, is it now fixed?
